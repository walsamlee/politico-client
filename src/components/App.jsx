import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import Landing from './Landing/Landing';
import Login from './containers/Login';
import Header from './Header/Header';
import Footer from './Footer/Footer';
import Parties from './containers/Parties';
import Register from './containers/Register';
import CreateOffice from './containers/CreateOffice';
import Candidates from './containers/Candidates';
import CreateParty from './containers/CreateParty';

const App = () => (
  <React.Fragment>
    <div className="main">
      <ToastContainer />
      <Header />
      <Switch>
        <Route exact path="/" component={Landing} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/parties" component={Parties} />
        <Route exact path="/register" component={Register} />
        <Route exact path="/create-office" component={CreateOffice} />
        <Route exact path="/candidates" component={Candidates} />
        <Route exact path="/create-party" component={CreateParty} />
      </Switch>
      <Footer />
    </div>
  </React.Fragment>
);

export default App;
