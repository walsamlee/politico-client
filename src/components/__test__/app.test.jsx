import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('<App />', () => {
  it('renders one <App /> component', () => {
    const component = shallow(<App />);
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
