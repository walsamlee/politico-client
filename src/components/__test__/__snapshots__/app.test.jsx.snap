// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders one <App /> component 1`] = `ShallowWrapper {}`;
